# Generated by Django 4.2.3 on 2024-08-27 10:27

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):
    dependencies = [
        ("user", "0006_user_company"),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ("ticket", "0011_ticket_solution"),
    ]

    operations = [
        migrations.CreateModel(
            name="KnownError",
            fields=[
                (
                    "basemodel_ptr",
                    models.OneToOneField(
                        auto_created=True,
                        on_delete=django.db.models.deletion.CASCADE,
                        parent_link=True,
                        primary_key=True,
                        serialize=False,
                        to="user.basemodel",
                    ),
                ),
                ("error", models.CharField(max_length=255)),
                ("solution", models.TextField()),
                (
                    "creaeted_by",
                    models.ForeignKey(
                        blank=True,
                        null=True,
                        on_delete=django.db.models.deletion.DO_NOTHING,
                        related_name="created_known_errors",
                        to=settings.AUTH_USER_MODEL,
                    ),
                ),
                (
                    "solution_ticket",
                    models.ForeignKey(
                        blank=True,
                        null=True,
                        on_delete=django.db.models.deletion.CASCADE,
                        related_name="known_errors",
                        to="ticket.ticket",
                    ),
                ),
            ],
            bases=("user.basemodel",),
        ),
    ]
